aerleon/aclcheck_cmdline.py

```diff
@@ -17,72 +17,104 @@
 """Command line interface to aclcheck library."""
 
 import pathlib
-from optparse import OptionParser
+from argparse import ArgumentParser, RawTextHelpFormatter
 
 from aerleon.lib import aclcheck, naming, policy, yaml
+from aerleon.utils import config
 
 
 def main():
-    # TODO(robankeny): Lets move this to gflags
     usage = 'usage: %prog [options] arg'
-    _parser = OptionParser(usage)
-    _parser.add_option(
+    _parser = ArgumentParser(
+        prog='aclcheck',
+        formatter_class=RawTextHelpFormatter,
+    )
+    _parser.add_argument(
         '--definitions-directory',
-        dest='definitions',
+        dest='definitions_directory',
         help='definitions directory',
-        default='./def',
     )
-    _parser.add_option(
+    _parser.add_argument(
         '--base-directory',
-        dest='basedir',
+        dest='base_directory',
         help='The base directory to look for include files.',
-        default='',
     )
-    _parser.add_option(
+    _parser.add_argument(
         '-p',
         '--policy-file',
         dest='pol',
         help='policy file',
-        default='./policies/pol/sample_paloalto_yaml.yaml',
+        required=True,
+    )
+    _parser.add_argument(
+        '--config-file',
+        dest='config_file',
+        help='config file',
     )
-    _parser.add_option(
-        '-d', '--destination', dest='dst', help='destination IP', default='200.1.1.1'
+    _parser.add_argument('-d', '--destination', dest='dst', help='destination IP')
+    _parser.add_argument(
+        '-s',
+        '--source',
+        dest='src',
+        help='source IP',
     )
-    _parser.add_option('-s', '--source', dest='src', help='source IP', default='any')
-    _parser.add_option(
+    _parser.add_argument(
         '--proto',
         '--protocol',
         dest='proto',
         help='Protocol (tcp, udp, icmp, etc.)',
-        default='tcp',
-    )
-    _parser.add_option(
-        '--dport', '--destination-port', dest='dport', help='destination port', default='80'
     )
-    _parser.add_option(
-        '--sport', '--source-port', dest='sport', help='source port', default='1025'
-    )
-    (FLAGS, unused_args) = _parser.parse_args()
+    _parser.add_argument('--dport', '--destination-port', dest='dport', help='destination port')
+    _parser.add_argument('--sport', '--source-port', dest='sport', help='source port')
+    FLAGS = _parser.parse_args()
+
+    default_flags = {
+        'base_directory': './policies',
+        'definitions_directory': './def',
+        'pol': None,
+        'config_file': None,
+        'dst': '200.1.1.1',
+        'src': 'any',
+        'proto': 'any',
+        'dport': '80',
+        'sport': '1025',
+    }
 
-    defs = naming.Naming(FLAGS.definitions)
-    with open(FLAGS.pol) as f:
+    configs = {}
+    configs.update(default_flags)
+
+    if not (FLAGS.base_directory and FLAGS.definitions_directory):
```

Comment:
If I am understanding this section correctly you are merging the command line flags onto the ones retrieved from the config file? But only in the case where base_directory and definition_directory are not set? Why not merge it always?

Reply:
aclcheck only uses the config file fields base_directory and definitions_directory. If the user provided both options on the command line we can skip reading the config file since the command line values take higher precedence.